### PR TITLE
Add kafka-jdbc-connector integration test

### PR DESCRIPTION
**Pull request by mkleen**
## Summary of the changes / Why this is an improvement

This adds an integration test to use the Kafka-Jdbc-Connector to insert data from Kafka intro CrateDB using the Postgres-Jdbc-Driver.

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: tests/kafka-connect-jdbc/run.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+docker-compose up -d
+
+echo "Wait until kafka-jdbc-connector is started up!"
+
+KAFKA_CONNECTOR_URL="localhost:8083/"
+
+timeout -s TERM 100 bash -c \
+'while [[ "$(curl -s -o /dev/null -L -w ''%{http_code}'' ${0})" != "200" ]];\
```
Comment (mfussenegger) [inline tests/kafka-connect-jdbc/run.sh]: Maybe the curl available already supports these options:

```
       --retry-connrefused
              In addition to the other conditions, consider ECONNREFUSED as a transient error too for --retry. This option is used together with --retry.

              Added in 7.52.0.

       --retry-delay <seconds>
              Make curl sleep this amount of time before each retry when a transfer has failed with a transient error (it changes the default backoff time algorithm between  re‐
              tries). This option is only interesting if --retry is also used. Setting this delay to zero will make curl use the default backoff time.

              If this option is used several times, the last one will be used.

              Added in 7.12.3.


       --retry-max-time <seconds>
              ...

       --retry <num>
```

File: tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        producer.send(toProducerRecord(schema, m3));
+        producer.send(toProducerRecord(schema, m4));
+        producer.send(toProducerRecord(schema, m5));
+
+        producer.flush();
+        producer.close();
+
+        Thread.sleep(5000);
+
+        var results = fetchResultsFromCrateDB();
```
Comment (mfussenegger) [inline tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java]: Maybe add an `assertBusy` method, similar to the one we have in CrateDB?

File: tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        var datumWriter = new ReflectDatumWriter<>(schema);
+        var outputStream = new ByteArrayOutputStream();
+
+        var encoder = EncoderFactory.get().binaryEncoder(outputStream, null);
+        datumWriter.write(metric, encoder);
+        encoder.flush();
+
+        var datumReader = new GenericDatumReader<GenericRecord>(schema);
+        var decoder = DecoderFactory.get().binaryDecoder(outputStream.toByteArray(), null);
+        return datumReader.read(null, decoder);
```
Comment (mfussenegger) [inline tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java]:
```suggestion
        try (var outputStream = new ByteArrayOutputStream()) {

          var encoder = EncoderFactory.get().binaryEncoder(outputStream, null);
          datumWriter.write(metric, encoder);
          encoder.flush();

          var datumReader = new GenericDatumReader<GenericRecord>(schema);
          var decoder = DecoderFactory.get().binaryDecoder(outputStream.toByteArray(), null);
          return datumReader.read(null, decoder);
        }
```

File: tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    }
+
+    void waitUntil(Supplier<Boolean> f) throws Exception {
+        for (int i = 1; i <= 10; i++) {
+            if (!f.get()) {
+                Thread.sleep(1000);
+            } else {
+                return;
+            }
+        }
```
Comment (mfussenegger) [inline tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java]: Maybe some exponentiall back-off could be used. But not sure if worth it - could help having faster tests on fast machines without overloading slow machines more.

File: tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        producerProps.put("bootstrap.servers", "127.0.0.1:9092");
+        producerProps.put("acks", "all");
+        producerProps.put("key.serializer", KafkaAvroSerializer.class.getName());
+        producerProps.put("value.serializer", KafkaAvroSerializer.class.getName());
+        producerProps.put("linger.ms", "10");
+        producerProps.put("schema.registry.url", "http://127.0.0.1:8081");
+        return new KafkaProducer<>(producerProps);
+    }
+
+    static final String PSQL_DRIVER = "jdbc:postgresql://localhost:54432/doc?user=crate";
```
Comment (mfussenegger) [inline tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java]: maybe move the constant to the top of the file

File: tests/kafka-connect-jdbc/run.sh (hunk 1 above)
Comment (mkleen) [reply tests/kafka-connect-jdbc/run.sh]: I first used `curl --retry`. It does not cover the case when a zero length response comes from the server and will stop, which happens here when to kafka-connector is starting up. This is why switch to this approach.

File: tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java (hunk 3 above)
Comment (mkleen) [reply tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java]: Good point, although there is nothing to close on a `ByteArrayOutputStream`.

File: tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java (hunk 2 above)
Comment (mkleen) [reply tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java]: 👍

File: tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java [hunk 6]
```diff
@@ -174,6 +173,38 @@ boolean isKafkaConnectorReady() {
         return new KafkaProducer<>(producerProps);
     }
 
-    static final String PSQL_DRIVER = "jdbc:postgresql://localhost:54432/doc?user=crate";
+    public static void assertBusy(CheckedRunnable<Exception> codeBlock, long maxWaitTime, TimeUnit unit) throws Exception {
```
Comment (mkleen) [inline tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java]: Currently it is copy-paste, maybe we could add this to crate-testing instead ?

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Pull request merged**